mtp_send_money/apps/send_money/forms.py

```
@@ -200,8 +209,52 @@ class DebitCardAmountForm(SendMoneyForm):
             'max_decimal_places': _('Only use 2 decimal places'),
         }
     )
+    error_messages = {
+        'connection': _('This service is currently unavailable'),
+        'cap_exceeded': _("The Prisoner's Account has reached its limit. \n Please try again in a few days.")
+    }
     serialise_amount = serialise_amount
     unserialise_amount = unserialise_amount
+    max_lookup_tries = 2
+    prisoner_capping_threshold_in_pence = settings.PRISONER_CAPPING_THRESHOLD_IN_PENCE
+    additional_fields_to_deserialize = ['prisoner_number']
+
+    @classmethod
+    def get_api_session(cls, reconnect=False):
+        with cls.shared_api_session_lock:
+            if reconnect or not cls.shared_api_session:
+                cls.shared_api_session = get_api_session()
+            return cls.shared_api_session
+
+    #  def clean(self):
+        #  try:
+            #  if not self.errors and not self.is_account_balance_below_threshold():
+                #  raise ValidationError(self.error_messages['cap_exceeded'], code='cap_exceeded')
+        #  except (RequestException, OAuth2Error):
+            #  logger.exception('Could not look up prisoner account balance')
+            #  raise ValidationError(self.error_messages['connection'], code='connection')
+        #  return self.cleaned_data
+
+    def is_account_balance_below_threshold(self):
+        prisoner_account_balance = self.lookup_prisoner_account_balance(
+            prisoner_number=self.data['prisoner_number']
+        )['combined_prisoner_account_balance']
+        logger.info('Prisoner account balance for %s is %s', self.data['prisoner_number'], prisoner_account_balance)
+        return prisoner_account_balance < self.PRISONER_CAPPING_THRESHOLD_IN_PENCE
```

this needs to be `<=` otherwise someone can't even reach the cap

---

My interpretation of the acceptance criteria was that they shouldn't be able to reach £900, only £899.99. https://dsdmoj.atlassian.net/browse/MTP-1465 Lets discuss it with the team tomorrow

---

the cap is definitely £900, not £899.99. in fact the story also says £1000, but that's wrong

---

also, the form variable is lower-cased so this line should throw an attribute error.
i think it's best to refer to the setting directly here rather than copying it onto the form class first – that way tests can override the setting using `override_settings`.